aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -301,23 +302,28 @@ void ImagingResampleVertical(
 // weights, but when aa=False they could be optimized further.
 template <typename scale_type, class F>
 void upsample_avx_bilinear_uint8(
-    const at::Tensor& input,
+    const at::Tensor& input_,
     const at::Tensor& output,
     bool align_corners,
     const scale_type& scales,
     bool antialias) {
-  auto batch_size = input.size(0);
-  auto num_channels = input.size(1);
-  auto xin = input.size(3);
-  auto yin = input.size(2);
+  auto batch_size = input_.size(0);
+  auto num_channels = input_.size(1);
+  auto xin = input_.size(3);
+  auto yin = input_.size(2);
   auto xout = output.size(3);
   auto yout = output.size(2);
 
   if (xin == xout && yin == yout) {
-    output.copy_(input);
+    output.copy_(input_);
     return;
   }
 
+  at::Tensor input = input_;
+  if (!(input.is_contiguous() || input.is_contiguous(at::MemoryFormat::ChannelsLast))) {
```

Comment:
Might be worth a comment, something like:

```suggestion
  // If the layout is "other", we explicitly convert the input to CL.
  // This simplifies the rest of the code and let us assume that the format is either CL or CF, nothing else.
  // Most tensors going through this `if` block won't need to go through unpacking, but those having C < 3 may
  // have to (this means 2 copies are made). We could avoid the extra copy by handling the "other" layout
  // directly within unpack_rgb() and pack_rgb(), but initial attempts showed that this is fairly complex.
  if (!(input.is_contiguous() || input.is_contiguous(at::MemoryFormat::ChannelsLast))) {
```